seaborn/_core/data.py

```diff
@@ -312,8 +312,18 @@ def convert_dataframe_to_pandas(data: object) -> pd.DataFrame:
         # in Plot.add(). But noting here in case this seems to be a bottleneck.
         return pd.api.interchange.from_dataframe(data)
     except Exception as err:
-        msg = (
-            "Encountered an exception when converting data source "
-            "to a pandas DataFrame. See traceback above for details."
-        )
-        raise RuntimeError(msg) from err
+        if hasattr(data, 'to_pandas'):
+            try:
+                return data.to_pandas()  # type: ignore[attr-defined]
```

Comment:
j/w but can type checkers not figure out that this attribute must exist in this context?